File: app/src/main/java/com/example/macarus0/walkiewalkie/data/DogOwnerDao.java
package com.example.macarus0.walkiewalkie.data;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;

import java.util.List;

@Dao
public abstract class DogOwnerDao {
    @Insert
    public abstract void insert(DogOwner... dogOwners);

    @Query("DELETE FROM DogOwner WHERE DogOwner.dogId = :dogId AND DogOwner.ownerId = :ownerId")
    public abstract void delete(long dogId, long ownerId);

}

File: app/src/main/java/com/example/macarus0/walkiewalkie/data/DogOwnerDao.kt
package com.example.macarus0.walkiewalkie.data

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query

@Dao
interface DogOwnerDao {
    @Insert
    fun insert(vararg dogOwners: DogOwner)

    @Query("DELETE FROM DogOwner WHERE DogOwner.dogId = :dogId AND DogOwner.ownerId = :ownerId")
    fun delete(dogId: Long, ownerId: Long)

    @Query("DELETE FROM DogOwner WHERE DogOwner.dogId = :dogId")
    fun deleteDog(dogId: Long)

    @Query("DELETE FROM DogOwner WHERE DogOwner.ownerId = :ownerId")
    fun deleteOwner(ownerId: Long)

}
